src/DomainHost.sol

```diff
@@ -0,0 +1,411 @@
+    function registerMint(TeleportGUID calldata teleport) external;
+}
+
+struct Settlement {
+    bytes32 sourceDomain;
+    bytes32 targetDomain;
+    uint256 amount;
+    bool    sent;
+}
+
+/// @title Support for xchain MCD, canonical DAI and Maker Teleport
+/// @dev This is just the business logic which needs concrete message-passing implementation
+abstract contract DomainHost {
+
+    // --- Data ---
+    mapping (address => uint256) public wards;
+    mapping (bytes32 => bool)    public teleports;
+    Settlement[]                 public settlementQueue;
+
+    bytes32     public immutable ilk;
+    VatLike     public immutable vat;
+    DaiJoinLike public immutable daiJoin;
+    DaiLike     public immutable dai;
+    address     public immutable escrow;
+    RouterLike  public immutable router;
+
+    address public vow;
+    uint256 public lid;         // Local ordering id
+    uint256 public rid;         // Remote ordering id
+    uint256 public line;        // Remote domain global debt ceiling [RAD]
+    uint256 public grain;       // Keep track of the pre-minted DAI in the escrow [WAD]
+    uint256 public sin;         // A running total of how much is required to re-capitalize the remote domain [WAD]
+    uint256 public cure;        // The amount of unused debt [RAD]
+    bool public cureReported;   // Returns true if cure has been reported by the guest
+    uint256 public live;
+
+    uint256 constant RAY = 10 ** 27;
+
+    string constant ARITHMETIC_ERROR = string(abi.encodeWithSignature("Panic(uint256)", 0x11));
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+    event Lift(uint256 wad);
+    event Release(uint256 wad);
+    event Push(int256 wad);
+    event Rectify(uint256 wad);
+    event Cage();
+    event Tell(uint256 value);
+    event Exit(address indexed usr, uint256 wad, uint256 claim);
+    event UndoExit(address indexed originalSender, uint256 wad);
+    event Deposit(address indexed to, uint256 amount);
+    event UndoDeposit(address indexed originalSender, uint256 amount);
+    event Withdraw(address indexed to, uint256 amount);
+    event RegisterMint(TeleportGUID teleport);
+    event InitializeRegisterMint(TeleportGUID teleport);
+    event FinalizeRegisterMint(TeleportGUID teleport);
+    event Settle(bytes32 sourceDomain, bytes32 targetDomain, uint256 amount);
+    event InitializeSettle(bytes32 sourceDomain, bytes32 targetDomain, uint256 amount);
+    event FinalizeSettle(bytes32 sourceDomain, bytes32 targetDomain, uint256 amount);
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "DomainHost/not-authorized");
+        _;
+    }
+
+    function _isGuest(address usr) internal virtual view returns (bool);
+    modifier guestOnly {
+        require(_isGuest(msg.sender), "DomainHost/not-guest");
+        _;
+    }
+
+    modifier vatLive {
```

Comment:
I think is just better to put `vat.live` and `live` checks directly in line. As it doesn't seem to be an homogeneous usage of them between the two domains.